### PR TITLE
fix: actor_id in submitting assessment and retrival of students result for mentor

### PR DESCRIPTION
**Pull request by Ajay-Maury**
Ticket:- https://github.com/Mission-Prerna/prerna-lakshya-app/issues/440

### CODE REVIEW

File: src/school/school.service.v2.ts [hunk 1]
```diff
@@ -96,18 +96,29 @@ export class SchoolServiceV2 extends SchoolService {
               COUNT(CASE WHEN a.is_passed = false THEN 1 END)                      AS failed,
               RANK() OVER (PARTITION BY a.student_id ORDER BY a.submitted_at DESC) AS rank
         FROM assessments a
+      LEFT JOIN mentor m ON a.mentor_id = m.id
         WHERE a.student_id IS NOT NULL
          AND a.udise = $1
          AND a.is_valid = true
-         AND a.actor_id = $2
-         AND a.student_id not in ('-1', '-2', '-3') --// we don't want anonymous students
+         AND (a.actor_id = $2 OR m.phone_no = $6)
```
Comment (tushar5526) [inline src/school/school.service.v2.ts:103]: why not drop a.actor_id completely ? `phone_no` is unique for every user.

File: src/school/school.service.v2.ts [hunk 2]
```diff
@@ -96,18 +96,29 @@ export class SchoolServiceV2 extends SchoolService {
               COUNT(CASE WHEN a.is_passed = false THEN 1 END)                      AS failed,
               RANK() OVER (PARTITION BY a.student_id ORDER BY a.submitted_at DESC) AS rank
         FROM assessments a
+      LEFT JOIN mentor m ON a.mentor_id = m.id
```
Comment (tushar5526) [inline src/school/school.service.v2.ts:99]: We are already joining on mentor_id here

File: src/school/school.service.v2.ts (hunk 1 above)
Comment (Ajay-Maury) [reply src/school/school.service.v2.ts:103]: Data is fetch on the basis actor_id bot for a particuler mentor, so need to keep it.

File: src/school/school.service.v2.ts (hunk 2 above)
Comment (Ajay-Maury) [reply src/school/school.service.v2.ts:99]: No any join with mentor needed

**Pull request merged**